mlflow/keras.py

```diff
@@ -429,11 +429,7 @@ def _load_model(model_path, keras_module, save_format, **kwargs):
     if save_format == "h5":
         model_path = model_path + ".h5"
 
-    from distutils.version import StrictVersion
-
-    if save_format == "h5" and StrictVersion(
-        keras_module.__version__.split("-")[0]
-    ) >= StrictVersion("2.2.3"):
+    if save_format == "h5" and Version(keras_module.__version__.split("-")[0]) >= Version("2.2.3"):
```

Comment:
seemingly we can directly compare version includes '-', like:
Version(keras_module.__version__) >= Version("2.2.3")

Reply:
Thanks for the catch!

Reply:
Found the PR that introduced the `split` logic here:

https://github.com/mlflow/mlflow/pull/1330

> Additionally, this adds a .split to the keras version check so it will work with "2.2.4-tf".

Reply:
Pushed the fix in https://github.com/mlflow/mlflow/pull/4386/commits/2edc90585d9291623f713079499c09fef5174911